### PR TITLE
Early stopping

### PR DESCRIPTION
**Pull request by AleAyotte**
Early stopping and learning rate decay added to Cnn Model and sub-classes.

### CODE REVIEW

File: Code/Model.py [hunk 1]
```diff
@@ -578,15 +617,14 @@ def fit(self, X_train=None, t_train=None, dtset=None, verbose=False, gpu=True):
         :param verbose: print the loss during training
         :param gpu: True: Train the model on the gpu. False: Train the model on the cpu
         """
+        
+        train_loader, valid_loader = self.set_train_valid_loader(X_train, t_train, dtset)
```
Comment (Rayn2402) [inline Code/Model.py:621]: I'm not sure why you're doing data splits (separates training and validation) in the fit function. I think this process should be done only in cross_validation function, which is using fit.

Comment (Rayn2402) [reply Code/Model.py:621]: Also, cross_validation function as a validation percentage parameter that avoid the need of an internal valid_size attribute for the class.

**Review by Rayn2402 — APPROVED** [review]

File: Code/Model.py (hunk 1 above)
Comment (Rayn2402) [reply Code/Model.py:621]: It's ok finally! I understand the way your code works and there is no conflict with the master.
